src/gui/FileBrowser.cpp

```diff
@@ -405,62 +510,71 @@ void FileBrowserTreeWidget::mousePressEvent(QMouseEvent * me )
 	FileItem * f = dynamic_cast<FileItem *>( i );
 	if( f != NULL )
 	{
-		m_pphMutex.lock();
-		if( m_previewPlayHandle != NULL )
-		{
-			Engine::mixer()->removePlayHandle(
-							m_previewPlayHandle );
-			m_previewPlayHandle = NULL;
-		}
+		previewFileItem(f);
+	}
+}
 
-		// in special case of sample-files we do not care about
-		// handling() rather than directly creating a SamplePlayHandle
-		if( f->type() == FileItem::SampleFile )
-		{
-			TextFloat * tf = TextFloat::displayMessage(
-					tr( "Loading sample" ),
-					tr( "Please wait, loading sample for "
-								"preview..." ),
-					embed::getIconPixmap( "sample_file",
-								24, 24 ), 0 );
-			qApp->processEvents(
-					QEventLoop::ExcludeUserInputEvents );
-			SamplePlayHandle * s = new SamplePlayHandle(
-								f->fullName() );
-			s->setDoneMayReturnTrue( false );
-			m_previewPlayHandle = s;
-			delete tf;
-		}
-		else if( ( f->extension ()== "xiz" || f->extension() == "sf2" || f->extension() == "sf3" || f->extension() == "gig" || f->extension() == "pat" ) &&
-			! pluginFactory->pluginSupportingExtension(f->extension()).info.isNull() )
-		{
-			m_previewPlayHandle = new PresetPreviewPlayHandle( f->fullName(), f->handling() == FileItem::LoadByPlugin );
-		}
-		else if( f->type() != FileItem::VstPluginFile &&
-				( f->handling() == FileItem::LoadAsPreset ||
-				f->handling() == FileItem::LoadByPlugin ) )
+
+
+
+void FileBrowserTreeWidget::previewFileItem(FileItem* file)
+{
+	m_pphMutex.lock();
+	if( m_previewPlayHandle != NULL )
+	{
+		Engine::mixer()->removePlayHandle(
+						m_previewPlayHandle );
+		m_previewPlayHandle = NULL;
+	}
+
+	// in special case of sample-files we do not care about
+	// handling() rather than directly creating a SamplePlayHandle
+	if( file->type() == FileItem::SampleFile )
+	{
+		TextFloat * tf = TextFloat::displayMessage(
+				tr( "Loading sample" ),
+				tr( "Please wait, loading sample for "
+							"preview..." ),
+				embed::getIconPixmap( "sample_file",
+							24, 24 ), 0 );
+		qApp->processEvents(
+				QEventLoop::ExcludeUserInputEvents );
+		SamplePlayHandle * s = new SamplePlayHandle( file->fullName() );
+		s->setDoneMayReturnTrue( false );
+		m_previewPlayHandle = s;
+		delete tf;
+	}
+	else if(
+		( file->extension ()== "xiz" || file->extension() == "sf2" ||
+		  file->extension() == "sf3" || file->extension() == "gig" ||
+		  file ->extension() == "pat" ) &&
+		! pluginFactory->pluginSupportingExtension(file->extension()).info.isNull() )
```

Comment:
I think that can be shortened to `! pluginFactory->pluginSupportingExtension(file->extension()).isNull() )` since the `PluginInfoAndKey` struct has a `isNull()` function returning `info.isNull()`.